### PR TITLE
[azure logs] Update docs with details about event hub name recommendations 

### PR DESCRIPTION
**Pull request by zmoog**
## What does this PR do?

Updates the Azure Logs documentation adding recommendations about the Event Hub name, and explaining how it is used by the integration when it contains underscores (_) characters.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] ~~I have verified that all data streams collect metrics or logs.~~
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] ~~I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~~

## Author's Checklist

- [ ]

## Related issues

- Relases #3006

## Screenshots

CleanShot 2022-05-05 at 17 07 05@2x

### CODE REVIEW

**Comment by zmoog** [comment]
Hey @ravikesarwani, here's the first edit of the Azure Logs documentation to inform the users about what characters to use for the Event Hub name to maximize compatibility.

I made some minor changes to the docs, but I want to know your perspective on how far we want to update the docs right now. 

The full details are in the files changed section, but I also added a screenshot with the rendering of the main README file.

CleanShot 2022-04-27 at 16 43 55@2x

**Review by ravikesarwani — COMMENTED** [review]
"Elastic recommends using only letters, numbers, and the hyphen (-) character for Event Hub names to maximize compatibility. You can use existing Event Hubs having underscores (_) in the name, and the integration will adjust the settings when required" This is great but leaves users wanting to know more. I think its okay to expand this a little more and explain the reason "why".

**Comment by zmoog** [comment]
> "Elastic recommends using only letters, numbers, and the hyphen (-) character for Event Hub names to maximize compatibility. You can use existing Event Hubs having underscores (_) in the name, and the integration will adjust the settings when required" This is great but leaves users wanting to know more. I think its okay to expand this a little more and explain the reason "why".

Yeah, I was puzzled about telling users more, or keeping things high level. 

Let's expand it a little, what about:

> It is a fully managed, real-time data ingestion service. Elastic recommends using only letters, numbers, and the hyphen (-) character for Event Hub names to maximize compatibility. You can use existing Event Hubs having underscores (_) in the Event Hub name. The integration will replace underscores with hyphens (-) when it uses the Event Hub name to create dependent Azure resources behind the scenes (e.g., the storage account container to store Event Hub consumer offsets).

**Comment by zmoog** [comment]
@ravikesarwani added a final rendering in the PR description: is there anything else you want to change in the Azure docs?

**Comment by ravikesarwani** [comment]
Good from my side.

File: packages/azure/_dev/build/docs/README.md [hunk 1]
```diff
@@ -53,32 +59,47 @@ The storage account key, this key will be used to authorize access to data in yo
...
+# Azure PublicCloud 
+https://management.azure.com/
+
+# Azure USGovernmentCloud
+https://management.usgovcloudapi.net/
+```
+
 Users can also use this in case of a Hybrid Cloud model, where one may define their own endpoints.
 
+## Logs
```
[the remark this answers is not in the document]
Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md:81]: Yeah, I was thinking about the same thing. Let me double-check if we have a doc for each existing data stream 👀

[the remark this answers is not in the document]
Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md:81]: @ravikesarwani do you agree with us? What's your thought?

[the remark this answers is not in the document]
Comment (ravikesarwani) [reply packages/azure/_dev/build/docs/README.md:81]: Yes, I am good with that and it makes it more consistent with AWS as well.
One thing I was thinking was to add a line like "For sample event and exported fields review documentation provided for each integration, example: Azure Activity Logs, Active Directory Logs (contains sign-in and audit logs), Azure Platform Logs, Azure Spring Cloud Logs etc."

[the remark this answers is not in the document]
Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md]: Updated!

[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply packages/azure/_dev/build/docs/README.md]: Thank you!

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**